source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/Calculation/CalculationOrchestrationActivitiesTests.cs

```diff
@@ -132,52 +133,38 @@ public async Task MockExternalDependencies_WhenCallingDurableFunctionEndPoint_Or
 
         var activities = completeOrchestrationStatus.History
             .OrderBy(item => item["Timestamp"])
-            .Select(item => item.Value<string>("FunctionName"));
+            .Select(item => item.ToObject<OrchestrationHistoryItem>())
+            .ToList();
 
         activities.Should().NotBeNull().And.Equal(
         [
-            "CalculationOrchestration",
-            "CreateCalculationRecordActivity",
-            "StartCalculationActivity",
-            "GetJobStatusActivity",
-            "UpdateCalculationStatusActivity",
-            "CreateCompletedCalculationActivity",
-            "SendCalculationResultsActivity",
-            null
+            new OrchestrationHistoryItem("ExecutionStarted", FunctionName: "CalculationOrchestration"),
+            new OrchestrationHistoryItem("TaskCompleted", FunctionName: "CreateCalculationRecordActivity"),
+            new OrchestrationHistoryItem("TaskCompleted", FunctionName: "StartCalculationActivity"),
+            new OrchestrationHistoryItem("TaskCompleted", FunctionName: "GetJobStatusActivity"),
+            new OrchestrationHistoryItem("TaskCompleted", FunctionName: "UpdateCalculationStatusActivity"),
+            new OrchestrationHistoryItem("TaskCompleted", FunctionName: "CreateCompletedCalculationActivity"),
+            new OrchestrationHistoryItem("TaskCompleted", FunctionName: "SendCalculationResultsActivity"),
+            new OrchestrationHistoryItem("TimerCreated"), // Wait for raised event (ActorMessagesEnqueued)
+            new OrchestrationHistoryItem("EventRaised", Name: "MessagesEnqueuedV1"),
+            new OrchestrationHistoryItem("TaskCompleted", FunctionName: "UpdateCalculationOrchestrationStateActivity"),
+            new OrchestrationHistoryItem("TaskCompleted", FunctionName: "UpdateCalculationOrchestrationStateActivity"),
+            new OrchestrationHistoryItem("ExecutionCompleted"),
```

Comment:
```suggestion
            new OrchestrationHistoryItem("ExecutionCompleted"),
```
If "OrchestrationHistoryItem" contained a nullable result, then you could delete line 156 -> 158 (var last = ...)